<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from rover.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="rover">
  <link name="world" />
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="-0.29581 0.2 0.041208"
        rpy="0 0 0" />
      <mass
        value="13.26" />
      <inertia
        ixx="0.24913"
        ixy="-5.2994E-07"
        ixz="0.0018499"
        iyy="0.63177"
        iyz="3.8882E-08"
        izz="0.85645" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="front_left">
    <inertial>
      <origin
        xyz="-5.5511E-17 0 0.051273"
        rpy="0 0 0" />
      <mass
        value="1.7624" />
      <inertia
        ixx="0.006565"
        ixy="-2.2795E-18"
        ixz="-3.5385E-19"
        iyy="0.006565"
        iyz="8.0979E-20"
        izz="0.010358" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover/meshes/front_left.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover/meshes/front_left.STL" />
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>100000.0</mu>
            <mu2>100000.0</mu2>
          </ode>
        </friction>
        <contact>
          <ode>
            <min_depth>0</min_depth>
          </ode>
        </contact>
      </surface>
    </collision>
  </link>
  <joint
    name="front_left_joint"
    type="continuous">
    <origin
      xyz="-0.07 0.5 0.01"
      rpy="1.5708 0 0" />
    <parent
      link="base_link" />
    <child
      link="front_left" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="front_right">
    <inertial>
      <origin
        xyz="-5.5511E-17 1.1102E-16 -0.051273"
        rpy="0 0 0" />
      <mass
        value="1.7624" />
      <inertia
        ixx="0.006565"
        ixy="-2.0349E-18"
        ixz="4.4753E-19"
        iyy="0.006565"
        iyz="-4.594E-19"
        izz="0.010358" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover/meshes/front_right.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover/meshes/front_right.STL" />
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>100000.0</mu>
            <mu2>100000.0</mu2>
          </ode>
        </friction>
        <contact>
          <ode>
            <min_depth>0</min_depth>
          </ode>
        </contact>
      </surface>
    </collision>
  </link>
  <joint
    name="front_right_joint"
    type="continuous">
    <origin
      xyz="-0.07 -0.1 0.01"
      rpy="1.5708 0 0" />
    <parent
      link="base_link" />
    <child
      link="front_right" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="rear_wheels">
    <inertial>
      <origin
        xyz="0 1.1102E-16 -0.3"
        rpy="0 0 0" />
      <mass
        value="3.8893" />
      <inertia
        ixx="0.026151"
        ixy="1.9018E-19"
        ixz="-2.652E-19"
        iyy="0.026151"
        iyz="-1.1256E-18"
        izz="0.020788" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover/meshes/rear_wheels.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover/meshes/rear_wheels.STL" />
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>100000.0</mu>
            <mu2>100000.0</mu2>
          </ode>
        </friction>
        <contact>
          <ode>
            <min_depth>0</min_depth>
          </ode>
        </contact>
      </surface>
    </collision>
  </link>
  <joint
    name="rear_wheel_joint"
    type="continuous">
    <origin
      xyz="-0.52 -0.1 0.01"
      rpy="1.5708 0 0" />
    <parent
      link="base_link" />
    <child
      link="rear_wheels" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link name="arn_base">
    <inertial>
      <origin rpy="0 0 0" xyz="0.092843 0.11115 0.19139"/>
      <mass value="1.1258"/>
      <inertia ixx="0.0046576" ixy="-4.2427E-10" ixz="-1.9958E-10" iyy="0.0013278" iyz="-0.00014449" izz="0.00438"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/arn_base.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/arn_base.STL"/>
      </geometry>
      
    </collision>
  </link>
  <link name="bicep">
    <inertial>
      <origin rpy="0 0 0" xyz="-9.5491E-08 0.050024 0.069997"/>
      <mass value="0.41299"/>
      <inertia ixx="0.0015642" ixy="-1.2115E-11" ixz="1.9733E-10" iyy="0.00097868" iyz="-0.00038761" izz="0.0008296"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/bicep.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/bicep.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="shoulder_joint" type="continuous">
    <origin rpy="1.5708 0.18321 0" xyz="0.092843 -0.11914 0.16839"/>
    <parent link="arn_base"/>
    <child link="bicep"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="bottom_wrist">
    <inertial>
      <origin rpy="0 0 0" xyz="-4.0905E-08 -0.02109 0.055"/>
      <mass value="0.73266"/>
      <inertia ixx="0.0015748" ixy="2.1317E-09" ixz="-1.5E-10" iyy="0.00084305" iyz="1.4705E-11" izz="0.0013045"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/bottom_wrist.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/bottom_wrist.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="bottom_wrist_joint" type="continuous">
    <origin rpy="0 0 2.3852" xyz="0 0.1 0.01"/>
    <parent link="bicep"/>
    <child link="bottom_wrist"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="elbow">
    <inertial>
      <origin rpy="0 0 0" xyz="-1.0046E-08 -0.10496 -0.01015"/>
      <mass value="0.4633"/>
      <inertia ixx="0.0010638" ixy="-4.2139E-12" ixz="2.3982E-10" iyy="0.00063995" iyz="-0.00019133" izz="0.00074255"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/elbow.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/elbow.STL"/>
      </geometry>
    </collision>
  </link>

  <joint name="elbow_joint" type="continuous">
    <origin rpy="0 -0.011377 0" xyz="0 -0.046157 0.055"/>
    <parent link="bottom_wrist"/>
    <child link="elbow"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="top_wrist">
    <inertial>
      <origin rpy="0 0 0" xyz="8.1204E-09 -0.02109 -0.0050001"/>
      <mass value="0.73266"/>
      <inertia ixx="0.0015748" ixy="2.8411E-09" ixz="-7.6201E-11" iyy="0.00084305" iyz="4.5363E-10" izz="0.0013045"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/top_wrist.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://rover/meshes/top_wrist.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="top_wrist_joint" type="continuous">
    <origin rpy="3.1416 0 3.1319" xyz="0 -0.15448 0"/>
    <parent link="elbow"/>
    <child link="top_wrist"/>
    <axis xyz="0 0 1"/>
  </joint>
  <!--######################################################################-->
  <joint name="arn_base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.15 0.395 0.063"/>
    <parent link="base_link"/>
    <child link="arn_base"/>
  </joint>
  <joint name="dummy_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.0 0.1"/>
    <parent link="world"/>
    <child link="base_link"/>
  </joint>

  <!--################################ ARM TRANSIMISSIONS ######################################-->
  <transmission name="shoulder_trans" type="SimpleTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="shoulder_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="shoulder_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="bottom_wrist_trans" type="SimpleTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="bottom_wrist_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="bottom_wrist_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="elbow_trans" type="SimpleTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="elbow_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="elbow_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="top_wrist_trans" type="SimpleTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="top_wrist_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="top_wrist_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
  </transmission>
   <!--################################ BODY TRANSIMISSIONS ######################################-->
   
     <transmission name="rear_drive_trans">
	   <type>transmission_interface/SimpleTransmission</type>
	   <joint name="rear_wheel_joint">
	     <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
	   </joint>
	   <actuator name="rear_motor">
	     <mechanicalReduction>1</mechanicalReduction>
	     <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
	   </actuator>
  </transmission>
  <transmission name="front_right_trans">
	   <type>transmission_interface/SimpleTransmission</type>
	   <joint name="front_right_joint">
	     <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
	   </joint>
	   <actuator name="f_r_motor">
	     <mechanicalReduction>1</mechanicalReduction>
	     <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
	   </actuator>
  </transmission>
  <transmission name="front_left_trans">
	   <type>transmission_interface/SimpleTransmission</type>
	   <joint name="front_left_joint">
	     <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
	   </joint>
	   <actuator name="f_l_motor">
	     <mechanicalReduction>1</mechanicalReduction>
	     <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
	   </actuator>
  </transmission>
  <!--######################################################################-->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>/</robotNamespace>
      <controlPeriod>0.001</controlPeriod>
      <legacyModeNS>false</legacyModeNS>
    </plugin>
  </gazebo>

</robot>
